### PR TITLE
cmd/gentypes: sync at DAP @ 1.47

### PR DESCRIPTION
**Pull request by hyangah**
https://github.com/microsoft/vscode-debugadapter-node/tree/049665a465bf74e5f78d648469acbef95f4b16fd

According to the spec, RestartArguments uses a union type (oneOf) and Go does not have a union type.
Both LaunchRequestArguments and AttachRequestArguments are represented as map[string]interface{},
so we may attempt to map this to map[string]interface{}. But if the spec starts to include other
types in the union, this will be a problem. For now, use interface{}.

Added -u and -o flags, and use them with the go generate tool.

### CODE REVIEW

**Review by polinasok — APPROVED** [review]

File: cmd/gentypes/gentypes.go [hunk 1]
```diff
@@ -479,9 +491,24 @@ var typesExcludeList = map[string]bool{
...
 
 func main() {
+	flag.Parse()
+
 	log.SetFlags(log.LstdFlags | log.Lshortfile)
 
-	inputFilename := os.Args[1]
+	if flag.NArg() != 1 {
+		fmt.Fprintln(os.Stderr, "Path to the DAP specification json file is required.")
+		fmt.Fprintln(os.Stderr, "go run cmd/gentypes/gentypes.go <path/to/debugProtocol.json>")
```
Comment (eliben) [inline cmd/gentypes/gentypes.go:500]: Nit: this printout will look odd if the user builds/installs a binary of this tool. Maybe use `os.Args[0]` or just something generic like `<gentypes> <path/...>`?

**Review by eliben — APPROVED** [review]

File: cmd/gentypes/gentypes.go (hunk 1 above)
Comment (hyangah) [reply cmd/gentypes/gentypes.go]: done

**Comment by hyangah** [comment]
Thanks @eliben and @polinasok 
What is the release schedule? Or is dlv supposed to pick up the pseudo version?

**Comment by polinasok** [comment]
There is no schedule. We do it as things accumulate. We add a new release tag here, and then submit a PR on the delve side like this: https://github.com/go-delve/delve/pull/2248.

**Comment by polinasok** [comment]
Added 0.5.0 tag and sent out https://github.com/go-delve/delve/pull/2460 to update he version in delve.

**Pull request merged**